### PR TITLE
CI: Add .gitattributes for PR reviews and clean .gitignore

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

* Trim .gitignore
* add .gitattribute to toggle by default some automatically generated files for PRs

**NOTE** couple of commits will demonstrate the behavior

## Component(s) name

`ci`

## How to test

will try a couple of dummy commit to see it working

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by ClausHolbechArista** [comment]
The trim of .gitignore is a bit too aggressive.
When checking out this PR locally, I see the following being detected as changes (meaning they are in my local filesystem but previously ignored:
image

- `_build` and `context` come from `ansible-builder`.
- The media files are probably some mkdocs artifacts.
- I am not sure where `venv` comes from, so maybe that should just be removed.

**Comment by gmuloc** [comment]
Is it some leftover that would never happen again if you were to remove your clone and clone again and use the repo normally?
I don't think our .gitignore should be tracking "legacy ignore"
are we still using `ansible-builder` ?

**Comment by gmuloc** [comment]
Same for the media for example playbook - can we find out how they are generated? I see we have a `_media` but not a `media` directory in there these days: https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/docs

are the files used anywhere? how do we generate them?


I can find them in two other tracked places though:
```
user% pwd
/Users/userr/Github/gmuloc/ansible-avd
userr% find . \( -name "example-playbook-deploy*" \)
./ansible_collections/arista/avd/media/example-playbook-deploy-eapi.gif
./ansible_collections/arista/avd/media/example-playbook-deploy-cvp.gif
./media/example-playbook-deploy-eapi.gif
./media/example-playbook-deploy-cvp.gif
```

**Comment by ClausHolbechArista** [comment]
ok, so the media is probably just leftover. I will remove that.
Ansible builder should still be possible using the "procedure" described here: https://github.com/aristanetworks/ansible-avd/blob/devel/execution-environment.yml#L5

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# Collapsing generated tables from PR diff by default
+/ansible_collections/arista/avd/roles/eos_cli_config_gen/tables/*  linguist-generated=true -merge -diff
+/ansible_collections/arista/avd/roles/eos_designs/tables/*  linguist-generated=true -merge -diff
+
+# Collapsing generated plugins doc from PR diff by default
+/ansible_collections/arista/avd/docs/plugins/*  linguist-generated=true
+
+# debatable maybe only for some scenarios ?
+# Collapsing molecule diffs by default for docs / structured_configs
+/ansible_collections/arista/avd/molecule/*/documentation/*  linguist-generated=true
```
Comment (gmuloc) [inline .gitattributes]:
```suggestion
# /ansible_collections/arista/avd/molecule/*/documentation/*  linguist-generated=true
```

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+# Collapsing generated tables from PR diff by default
+/ansible_collections/arista/avd/roles/eos_cli_config_gen/tables/*  linguist-generated=true -merge -diff
+/ansible_collections/arista/avd/roles/eos_designs/tables/*  linguist-generated=true -merge -diff
+
+# Collapsing generated plugins doc from PR diff by default
+/ansible_collections/arista/avd/docs/plugins/*  linguist-generated=true
+
+# debatable maybe only for some scenarios ?
+# Collapsing molecule diffs by default for docs / structured_configs
```
Comment (gmuloc) [inline .gitattributes]:
```suggestion
```

File: .gitattributes [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Collapsing generated schemas from PR diff by default
+/ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml  linguist-generated=true
+/ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.jsonschema.json  linguist-generated=true
+/ansible_collections/arista/avd/roles/eos_designs/schemas/eos_designs.schema.yml  linguist-generated=true
+/ansible_collections/arista/avd/roles/eos_designs/schemas/eos_designs.jsonschema.json  linguist-generated=true
```
Comment (gmuloc) [inline .gitattributes]:
```suggestion
/ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml  linguist-generated=true -merge -diff
/ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.jsonschema.json  linguist-generated=true  -merge -diff
/ansible_collections/arista/avd/roles/eos_designs/schemas/eos_designs.schema.yml  linguist-generated=true  -merge -diff
/ansible_collections/arista/avd/roles/eos_designs/schemas/eos_designs.jsonschema.json  linguist-generated=true  -merge -diff
```

File: .gitattributes [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+# Collapsing generated plugins doc from PR diff by default
+/ansible_collections/arista/avd/docs/plugins/*  linguist-generated=true
+
+# debatable maybe only for some scenarios ?
+# /ansible_collections/arista/avd/molecule/*/documentation/*  linguist-generated=true
+
+# Excluding docs from stats for linguist
+/ansible_collections/arista/avd/roles/*/docs/*  linguist-documentation
+/ansible_collections/arista/avd/docs/*  linguist-documentation
+/ansible_collections/arista/avd/docs/*  linguist-documentation
```
Comment (gmuloc) [inline .gitattributes]:
```suggestion
```

**Review by ClausHolbechArista — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Comment by JulioPDX** [comment]
@ClausHolbechArista @gmuloc The venv comes from using python virtual environments. Great if you use a different virtual env per project.

**Pull request merged**